### ISSUE TITLE
errors when seeding with ECO

### ISSUE THREAD

**Issue by srobb1** [INCATools/ontology-development-kit#108]
Hello,

I am using the most up-to-date version of ODK (pulled today).

When I try to seed a new repository with ECO I get an error. If I leave off the ECO it runs fine.  

I suspect that there are specific issues with ECO that are are not playing well with ODK. Maybe not passing some of the rules. Should I post in ECO repo as well? As a temporary solution Is there another way to include terms from ECO without seeding with it?

Command:
```
./seed-via-docker.sh  -d ro pato eco -u srobb1 -t "Apple Snail Expression Pattern Ontology" ase
```
Error:
```
2018-09-27 19:01:43,945 ERROR org.obolibrary.obo2owl.OWLAPIObo2Owl - error executing obo id to IRI
com.google.common.util.concurrent.UncheckedExecutionException: java.lang.NullPointerException
	at com.google.common.cache.LocalCache$Segment.get(LocalCache.java:2203)
	at com.google.common.cache.LocalCache.get(LocalCache.java:3937)
	at com.google.common.cache.LocalCache.getOrLoad(LocalCache.java:3941)
	at com.google.common.cache.LocalCache$LocalLoadingCache.get(LocalCache.java:4824)
	at org.obolibrary.obo2owl.OWLAPIObo2Owl.oboIdToIRI(OWLAPIObo2Owl.java:1664)
	at org.obolibrary.oboformat.writer.OBOFormatWriter$OWLOntologyNameProvider.getName(OBOFormatWriter.java:1182)
	at org.semanticweb.owlapi.oboformat.OBOFormatRenderer$1.getName(OBOFormatRenderer.java:74)
	at org.obolibrary.oboformat.writer.OBOFormatWriter.write(OBOFormatWriter.java:325)
	at org.obolibrary.oboformat.writer.OBOFormatWriter.write(OBOFormatWriter.java:203)
	at org.semanticweb.owlapi.oboformat.OBOFormatRenderer.render(OBOFormatRenderer.java:88)
	at org.semanticweb.owlapi.oboformat.OBOFormatStorer.storeOntology(OBOFormatStorer.java:42)
	at org.semanticweb.owlapi.util.AbstractOWLStorer.store(AbstractOWLStorer.java:99)
	at org.semanticweb.owlapi.util.AbstractOWLStorer.storeOntology(AbstractOWLStorer.java:64)
	at uk.ac.manchester.cs.owl.owlapi.OWLOntologyManagerImpl.saveOntology(OWLOntologyManagerImpl.java:1302)
	at org.obolibrary.robot.IOHelper.saveOntology(IOHelper.java:559)
	at org.obolibrary.robot.IOHelper.saveOntology(IOHelper.java:500)
	at org.obolibrary.robot.ConvertCommand.execute(ConvertCommand.java:167)
	at org.obolibrary.robot.CommandManager.executeCommand(CommandManager.java:246)
	at org.obolibrary.robot.CommandManager.execute(CommandManager.java:190)
	at org.obolibrary.robot.CommandManager.main(CommandManager.java:139)
	at org.obolibrary.robot.CommandLineInterface.main(CommandLineInterface.java:53)
Caused by: java.lang.NullPointerException
	at org.obolibrary.obo2owl.OWLAPIObo2Owl.translateShorthandIdToExpandedId(OWLAPIObo2Owl.java:1766)
	at org.obolibrary.obo2owl.OWLAPIObo2Owl.oboIdToIRI_load(OWLAPIObo2Owl.java:1705)
	at org.obolibrary.obo2owl.OWLAPIObo2Owl$1.load(OWLAPIObo2Owl.java:81)
	at org.obolibrary.obo2owl.OWLAPIObo2Owl$1.load(OWLAPIObo2Owl.java:77)
	at com.google.common.cache.LocalCache$LoadingValueReference.loadFuture(LocalCache.java:3527)
	at com.google.common.cache.LocalCache$Segment.loadSync(LocalCache.java:2319)
	at com.google.common.cache.LocalCache$Segment.lockedGetOrLoad(LocalCache.java:2282)
	at com.google.common.cache.LocalCache$Segment.get(LocalCache.java:2197)
	... 20 more
Use the -vvv option to show the stack trace.
Use the --help option to see usage information.
Makefile:51: recipe for target 'ase.obo' failed
make: *** [ase.obo] Error 1
rm imports/pato_terms_combined.txt imports/eco_terms_combined.txt imports/ro_terms_combined.txt
ERROR RUNNING: cd src/ontology && make prepare_release
Exiting. Run with '-f' to force execution and ignore errors at ./seed-my-ontology-repo.pl line 330.
```

Thanks,
Sofia

**Comment by rctauber** [comment]
Hi @srobb1 - this is an issue converting the full `eco.owl` file to OBO. The converter enforces pretty strict rules for the OBO format and the OWL format violates some of these. We offer an [`eco-basic.owl`](http://purl.obolibrary.org/obo/eco/eco-basic.owl) file to get around this.

Trying this with `-d eco/eco-basic` causes an issue because it puts the import file in another folder within the `imports` folder, when the ODK just expects all the imports to be individual files in that folder. @cmungall - do you think the ODK could support dependencies like this?

**Comment by cmungall** [comment]
We want everything to starting using the base files for making import modules, where available (#50), e.g.
http://purl.obolibrary.org/obo/eco/eco-base.owl
(should be registered on http://obofoundry.org/ontology/eco.html ...)

Note that this violates the obo format comment cardinality restriction; rather than making the combination of base x basic we should just make odk use the `--no-check true` option when writing obo

**Comment by matentzn** [comment]
-> I also added --check false now; so it should work either way (base or otherwise).
https://github.com/INCATools/ontology-development-kit/issues/96

Will close this, please re-open if it happens again :)
